### ISSUE TITLE
Wifi connection showing not connected - even when it is 

### ISSUE THREAD

**Issue by Ultraviolet93** [SpyGuard/SpyGuard#31]
I've successfully installed Spyguard on RaspOs Bookworm on  Raspberry Pi 4.

All works fine but it shows as Wifi not connected at the bottom right of the capture screen.

However, it is connected, scans and creates reports correctly.

I've tried swapping round the on board WiFi with usb dongle for

AP and Internet link, but it makes no difference in either configuration.

Everything looks correct in the new network manager for bookworm sudo nitui.

Anyone else seen this issue  and found a solution ? 

![image](https://github.com/SpyGuard/SpyGuard/assets/157704007/01e69828-40ac-4089-a728-989c40b15871)

**Comment by icedan** [comment]
OS? nmcli installed?

**Comment by Ultraviolet93** [comment]
Thank you for the suggestions - I have nmcli installed which was installed prior to installing Spyguard. 

Hardware 
---------------
Raspberry Pi 4 Model B Rev 1.4

OS
----

PRETTY_NAME="Debian GNU/Linux 12 (bookworm)"
NAME="Debian GNU/Linux"
VERSION_ID="12"
VERSION="12 (bookworm)"
VERSION_CODENAME=bookworm
ID=debian
Linux 6.1.0-rpi7-rpi-v8 #1 SMP PREEMPT Debian 1:6.1.63-1+rpt1 (2023-11-24) aarch64 GNU/Linux


eth0 and lo network Settings 
------------------------------------------ 

eth0: flags=4099<UP,BROADCAST,MULTICAST>  mtu 1500
        ether e4:5f:01:63:77:07  txqueuelen 1000  (Ethernet)
        RX packets 0  bytes 0 (0.0 B)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 0  bytes 0 (0.0 B)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

lo: flags=73<UP,LOOPBACK,RUNNING>  mtu 65536
        inet 127.0.0.1  netmask 255.0.0.0
        inet6 ::1  prefixlen 128  scopeid 0x10<host>
        loop  txqueuelen 1000  (Local Loopback)
        RX packets 55405  bytes 5091882 (4.8 MiB)
        RX errors 0  dropped 0  overruns 0  frame 0
        TX packets 55405  bytes 5091882 (4.8 MiB)
        TX errors 0  dropped 0 overruns 0  carrier 0  collisions 0

Not sure if anything jumps out in the above ?  I have swapped over wlan0 and wlan1 but that doesn't seem to make any difference. The scan and reports work in either of these configurations although I understand the preferred arrangement is wlan1 for internet and wlan0 for AP which is how I've got it set up ?

**Comment by CSerpent** [comment]
Sorry if this is looping the same question/answer as above, but have you gone into raspi-config (opt 5 then AA from memory) and changed the network handler to nmcli?  This one got me repeatedly.

**Comment by CSerpent** [comment]
Yeah the raspi-config is different in bookworm.  Out of interest, do you have a USB WiFi stick you can plug in and see if you get the same issue?

**Comment by Ultraviolet93** [comment]
Good News !  I've reinstalled Spyguard and it now appears to be working !
Thank you for the suggestion - I should have tried that first ! 

The only weird thing is if I use wlan0 as the access point and wlan1 for the internet connection it fails to create the access point. 

If  I reverse the adapters in the backend  its works fine. Not sure why its behaving that way. But at least its working with Access point wlan1 and internet wlan0 . 

Be nice to understand what causes that issue but at least its up and running. 

Thank you

**Comment by Ultraviolet93** [comment]
Just checked the syslog , and yes I see   _nl80211 driver interface is not designed to be used with ap_scan=2_  message. 

I guess this is an issue with the particular usb wifi dongle I'm using ( RALINK 5370  802.11N)  I've also tried an Asus N150 -USB N10 Nano. 

The RALINK will allow devices to connect and run scans but only when AP is set to Ralink adapter and Internet to onboard Wi-Fi (atheros 8495)  

The ASUS creates the AP but nothing will connect,

I'm guessing this is a driver issue? Anyway to fix it or will I need to get hold of specific usb wifi adapter ?

**Comment by b068931cc450442b63f5b3d276ea4297** [comment]
I'll have a look at it in the next few days and then answer you here.
